test/runtests.jl

```diff
@@ -627,6 +627,7 @@ end
 
     @test Eye{Int}(∞) * D ≡ Eye{Int}(∞) * D ≡ D
     @test Eye(∞) * D == Eye(∞) * D == D
+    @test Eye(∞) == Eye(∞)^0 == Eye(∞)^1
```

Comment:
Add tests for `one` and `copy` directly as well.

Reply:
Add tests that `Eye(∞) == Eye(∞)^2`

Reply:
Add tests for `Diagonal(Fill(2,∞))`

Reply:
`^2` shouldn't be a problem because `LinearAlgebra` deals with matrix powers like this:
```julia
# base\intfuncs.jl:243
function power_by_squaring(x_, p::Integer)
    x = to_power_type(x_)
    if p == 1
        return copy(x)
    elseif p == 0
        return one(x)
    elseif p == 2
        return x*x
    elseif p < 0
        isone(x) && return copy(x)
        isone(-x) && return iseven(p) ? one(x) : copy(x)
        throw_domerr_powbysq(x, p)
    end
    t = trailing_zeros(p) + 1
    p >>= t
    while (t -= 1) > 0
        x *= x
    end
    y = x
    while p > 0
        t = trailing_zeros(p) + 1
        p >>= t
        while (t -= 1) >= 0
            x *= x
        end
        y *= x
    end
    return y
end
```